src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -308,7 +315,8 @@ public void insert(SinkRecord record) {
     }
 
     TopicPartitionChannel channelPartition = partitionsToChannel.get(partitionChannelKey);
-    channelPartition.insertRecordToBuffer(record);
+    boolean isFirstPartitionRowInBatch = channelsVisitedPerBatch.add(partitionChannelKey);
```

Comment:
if this batch has records containing two different partitions. 
This will return true twice. 

the naming seems misleading IMO, shouldnt it be something like `isFirstRowForPartition`?